tests/regression_tests/source_dlopen/inputs_true.dat

```diff
@@ -18,6 +18,6 @@
     <particles>1000</particles>
     <batches>10</batches>
     <inactive>0</inactive>
-    <source library="build/libsource.so" strength="1.0"/>
+    <source strength="1.0" type="source"/>
```

Comment:
This source should still have a `'library'` member, if I understand the changes to `source.py` correctly.

Reply:
Good catch! I'll look into that.

Reply:
It turns out that we were writing a default `IndependentSource` because the `library` attribute was being added after creation of an `IndepdendentSource` instance:

```python
source = openmc.IndependentSource()
source.library = 'libcustomsource.so`
```

I've re-added the `library`, `file`, and `parameters` attributes to the `IndependentSource` class so they return `AttributeErrors` when referenced. This should provide clarity to users on how to update their file and compiled sources. The downside is that those attribute names are now occupied, which is somewhat unusual, but I think the pros of directing users on how to update their Python code outweigh the cons in this case.

Reply:
Also added some tests to make sure these errors appear in the constructor or if the attributes are set later on.